packages/migrate/src/migrations/v0.41/typespec-rename.ts

```diff
@@ -0,0 +1,200 @@
+import { getAnyExtensionFromPath, NodePackage } from "@typespec/compiler";
+import type { CadlScriptNode, Node } from "@typespec/compiler-v0.40";
+import * as path from "path";
+import type { TypeSpecCompilerV0_40 } from "../../migration-config.js";
+import {
+  ContentMigrateAction,
+  createContentMigration,
+  createFileRenameMigration,
+  createPackageVersionMigration,
+  fileRenameAction,
+  MigrationContext,
+  MigrationKind,
+  packageVersionUpdateAction,
+} from "../../migration-types.js";
+
+export const updatePackageVersion = createPackageVersionMigration({
+  name: "Update package version",
+  kind: MigrationKind.PackageVersionUpdate,
+  migrate: (pkg: NodePackage) => {
+    const actions: Array<packageVersionUpdateAction> = [];
+
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@cadl-lang/compiler",
+      renamePackageName: "@typespec/compiler",
+      toVersion: "0.41.0",
+    });
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@cadl-lang/openapi",
+      renamePackageName: "@typespec/openapi",
+      toVersion: "0.41.0",
+    });
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@cadl-lang/openapi3",
+      renamePackageName: "@typespec/openapi3",
+      toVersion: "0.41.0",
+    });
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@cadl-lang/http",
+      renamePackageName: "@typespec/http",
+      toVersion: "0.41.0",
+    });
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@cadl-lang/versioning",
+      renamePackageName: "@typespec/versioning",
+      toVersion: "0.41.0",
+    });
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@cadl-lang/rest",
+      renamePackageName: "@typespec/rest",
+      toVersion: "0.41.0",
+    });
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@cadl-lang/lint",
+      renamePackageName: "@typespec/lint",
+      toVersion: "0.41.0",
+    });
+    actions.push({
+      kind: MigrationKind.PackageVersionUpdate,
+      packageName: "@azure-tools/cadl-autorest",
```

Comment:
Here it would be nice to be able to add migration actions, so that we could define a migration based on this one in typespec-azure, rather than encoding Azure packages here

Reply:
Agreed. I would like it to be completely data-driven for these predefined types of migrations to make this package completely generic and easier. This may solve our OSS/Azure split problem. Given the timeline, I would like to push this out to later.